#!/usr/bin/env python
PACKAGE = "masterslave"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
trajectory_enum = gen.enum([gen.const("PTP", int_t, 0, "PTP Trajectory"),
			    gen.const("Line", int_t, 1, "straight line trajectory"),
			    gen.const("Circle", int_t, 2, "circle trajectory")],
			    "an enum to set trajectory type")
gen.add("curTrajType",int_t,0, "Current Trajectory Type", 0, 0, 2, edit_method=trajectory_enum)
gen.add("Start",bool_t,0,"Start of the Trajectory", False)
gen.add("LengthX", int_t,0,"Length of the straight line", 0, 1, 100)
gen.add("LengthY", int_t,0,"Length of the straight line", 0, 1, 100)
gen.add("Radius",int_t,0,"Radius of the Circle", 1, 1, 50)
gen.add("zCoord",int_t,0,"z-Coordinate of the Circle",13,5,80)
gen.add("Speed", double_t,0,"Speed in mm/s",0.1, 0.1, 10)
exit(gen.generate(PACKAGE, "trajectorygenerator", "trajectorygenerator"))

